modules/howtos/pages/provisioning-cluster-resources.adoc

```diff
@@ -150,34 +152,82 @@ include::example$CollectionManagerExample.java[tag=scopeAdmin, indent=0]
+
+The example below shows how to create a simple primary index, restricted to a named scope and collection, by calling the `createPrimaryIndex()` method.
+Note that you cannot provide a named scope or collection separately, both must be set for the `QueryIndexManager` to create an index on the relevant keyspace path. 
+
+.Creating a primary index
+
+[source,java]
+----
+include::example$QueryIndexManagerExample.java[tag=primary,indent=0]
+----
+
+When a primary index name is not specified, the SDK will create the index as `#primary` by default.
+However, if you wish to provide a custom name, you can simply set an `indexName` property in the `CreatePrimaryQueryIndexOptions` object.
+
+You may have noticed that the example also sets the `ignoreIfExists` boolean flag. 
+When set to `true`, this optional argument ensures that an exception is not thrown if an index under the same name already exists.
+
+Creating a _secondary_ index follows a similar approach, with some minor differences:
+
+.Creating a secondary index
+
+[source,java]
+----
+include::example$QueryIndexManagerExample.java[tag=secondary,indent=0]
+----
 
-* QueryIndexManager -- https://docs.couchbase.com/sdk-api/couchbase-java-client/com/couchbase/client/java/manager/query/QueryIndexManager.html[`QueryIndexManager(AsyncQueryIndexManager async)`];
-* AnalyticsIndexManager -- https://docs.couchbase.com/sdk-api/couchbase-java-client/com/couchbase/client/java/manager/analytics/AnalyticsIndexManager.html[`AnalyticsIndexManager(Cluster cluster)`];
-* SearchIndexManager -- https://docs.couchbase.com/sdk-api/couchbase-java-client/com/couchbase/client/java/manager/search/SearchIndexManager.html[`SearchIndexManager(AsyncSearchIndexManager asyncIndexManager)`];
-* ViewIndexManager -- https://docs.couchbase.com/sdk-api/couchbase-java-client/com/couchbase/client/java/manager/view/ViewIndexManager.html[`ViewIndexManager(AsyncViewIndexManager async)`].
+The `createIndex()` method requires an index name to be provided, along with the fields to create the index on.
+Like the _primary_ index, you can restrict a _secondary_ index to a named scope and collection by passing some options.
 
-// * Query
+Indexes can easily take a long time to build if they contain a lot of documents.
+In these situations, it is more ideal to build indexes in the background.
+To achieve this we can use the `deferred` boolean option, and set it to `true`.
+
+.Deferring index creation
+
+[source,java]
+----
+include::example$QueryIndexManagerExample.java[tag=defer-indexes,indent=0]
+----
+
+To delete a query index you can use the `dropIndex()` or `dropPrimaryIndex()` methods. 
+Which one you use depends on the type of query index you wish to drop from the database.
+
+.Deleting an index
+
+[source,java]
+----
+include::example$QueryIndexManagerExample.java[tag=drop-primary-or-secondary-index,indent=0]
+----
 
 // * Search - note & link to FTS page & API?
 
 == View Management
```

Comment:
I know this was already there - but should we change it to Views Management to be consistent?